src/Workspaces/Core/Portable/Recommendations/AbstractRecommendationService.cs

```diff
@@ -93,9 +95,106 @@ internal abstract class AbstractRecommendationService : IRecommendationService
                 Concat(symbols.WhereAsArray(s => s.Kind != SymbolKind.Field));
         }
 
-        private static bool IsFriendlyName(int i, string elementName)
+        protected static ImmutableArray<ISymbol> LookupSymbolsInContainer(
+            INamespaceOrTypeSymbol container,
+            SemanticModel semanticModel,
+            int position,
+            bool excludeInstance)
         {
-            return elementName != null && string.Compare(elementName, "Item" + (i + 1), StringComparison.OrdinalIgnoreCase) != 0;
+            return excludeInstance
+                    ? semanticModel.LookupStaticMembers(position, container)
+                    : SuppressDefaultTupleElements(container,
+                        semanticModel.LookupSymbols(position, container, includeReducedExtensionMethods: true));
+        }
+
+        protected static ImmutableArray<ISymbol> GetSymbols(
+            SemanticModel semanticModel,
+            INamespaceOrTypeSymbol container,
+            int position,
+            bool useBaseReferenceAccessibility,
+            bool excludeInstance)
+        {
+            return useBaseReferenceAccessibility
+                ? semanticModel.LookupBaseMembers(position)
+                : LookupSymbolsInContainer(container, semanticModel, position, excludeInstance);
+        }
+
+        protected static ImmutableArray<ISymbol> GetSymbols<TLambdaExpressionSyntax, TArgumentSyntax, TArgumentListSyntax, TInvocationExpressionSyntax>(
+            SemanticModel semanticModel,
+            IParameterSymbol parameter,
+            int position,
+            Func<TArgumentListSyntax, SeparatedSyntaxList<TArgumentSyntax>> getArguments,
+            CancellationToken cancellationToken)
```

Comment:
whooboy.  this is complex.  

1. Consider doc'ing this.
2. should we pull these type parameters up to the class?  Does either language ever call this same method with a *different* instantiation?